drivers/dma/dma_iproc_pax.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Broadcom
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef DMA_IPROC_PAX
+#define DMA_IPROC_PAX
+
+#include "dma_iproc_pax_common.h"
+
+/* Register RM_CONTROL fields */
+#define RM_CONTROL_MODE_MASK 0x3
+#define RM_CONTROL_MODE_SHIFT 0
+#define RM_CONTROL_MODE_TOGGLE 0x2
+#define RM_CONTROL_CONFIG_DONE BIT(2)
+#define RM_CONTROL_CONFIG_DONE_SHIFT 2
+#define RM_CONTROL_CONFIG_DONE BIT(4)
+#define RM_CONTROL_LINE_INTR_EN_SHIFT 4
+#define RM_CONTROL_LINE_INTR_EN BIT(4)
+#define RM_CONTROL_AE_TIMEOUT_EN_SHIFT 5
+#define RM_CONTROL_AE_TIMEOUT_EN BIT(5)
+
+#define TYPE_DMA_DESC 0x3
+#define NUM_BD_BUFFS 8
+/* DMA desc count: 3 entries per packet */
+#define RM_DESC_BDCOUNT 3
+/* 1 DMA packet desc takes 3 BDs */
+#define DMA_DESC_SIZE (RM_DESC_BDWIDTH * RM_DESC_BDCOUNT)
+/* Max size of transfer in single packet */
+#define MAX_DMA_SIZE_PER_BD (16 * 1024 * 1024)
```

Comment:
what is the calculation behind the 16MBytes limits?

Reply:
Hi,
In V1 RM version, length field is of 25 bits in DMA Header descriptor.
So hardware can support upto 32MB of data length per BD, but we have set it to 16MB.

In case of V2 RM version, length field  in src/dst descriptor is modified to 15 bits.
So with this it can support max length of 32KB per BD.

But we do have an option to use Mega SRC and Mega DST which will support 16*LENGTH bits, so with this max 16*32KB = 512KB.